packages/style-engine/class-wp-style-engine.php

```diff
@@ -396,32 +416,19 @@ public function generate( $block_styles, $options ) {
 		}
 
 		// Build CSS rules output.
-		$css_selector              = isset( $options['selector'] ) ? $options['selector'] : null;
-		$filtered_css_declarations = array();
-
-		if ( ! empty( $css_declarations ) ) {
-			// Generate inline style declarations.
-			foreach ( $css_declarations as $css_property => $css_value ) {
-				$filtered_css_declaration = esc_html( safecss_filter_attr( "{$css_property}: {$css_value}" ) );
-				if ( ! empty( $filtered_css_declaration ) ) {
-					$filtered_css_declarations[] = $filtered_css_declaration . ';';
-				}
-			}
-		}
+		$css_selector = isset( $options['selector'] ) ? $options['selector'] : null;
+		$style_rules  = $this->get_css_rules_object( $options['selector'] );
+		$style_rules->add_declarations( $css_declarations );
```

Comment:
I tried to keep the `WP_Style_Engine_Rule` object strictly for rules... which means that the rules object is not aware of the `selector`, it's just for rules. That makes it simpler to keep things organized since the "assigning" of rules to selectors _should_ preferably be done in a `store` object.
Keeping it like this will make future implementations like a store, processing/combining styles etc easier 😉

Reply:
> That makes it simpler to keep things organized since the "assigning" of rules to selectors should preferably be done in a store object.

Ah, got it. Thanks for explaining.